systems/api/api-gateway/pkg/client/client.go

```diff
@@ -0,0 +1,317 @@
+package client
```

Comment:
We have some similar kind of code repetition in clients, Where we Marshal req, send req and unmarshall response.
We had the similar structure in Testing system, I added a function to abstract that part. Have a look
https://github.com/ukama/ukama/blob/ed21058a74e0b7d8ab03cf9eedd0d017133336c8/testing/integration/pkg/utils/rest.go#L211-L300

And that's how we can call it

https://github.com/ukama/ukama/blob/ed21058a74e0b7d8ab03cf9eedd0d017133336c8/testing/integration/pkg/nucleus/nucleus.go#L53-L62

Reply:
As discussed, we'll investigate the better way to perform this refactoring in another issue, so that it can benefit other systems.